### PR TITLE
promoted tuples

### PR DESCRIPTION
**Pull request by ersran9**

### CODE REVIEW

**Comment by judah** [comment]
Thank you for the change!  This looks good to me.

I had to reconfigure CircleCI to allow it to run on PRs of forks.  Can you please push a trivial change (e.g., `git commit --allow-empty`) and see if that triggers it correctly?

**Comment by ersran9** [comment]
hmm its failing for ghc-8.2.x and ghc 8.4.x. Would it be okay to wrap it with `#ifdef` s?

File: src/GHC/SourceGen/Syntax/Internal.hs [hunk 1]
```diff
@@ -88,6 +91,9 @@ noExtOrPlaceHolder = withPlaceHolder
 withPlaceHolder :: (PlaceHolder -> a) -> a
 withPlaceHolder = ($ PlaceHolder)
 
+withPlaceHolders :: ([PlaceHolder] -> a) -> a
+withPlaceHolders = ($ [])
```
Comment (ersran9) [inline src/GHC/SourceGen/Syntax/Internal.hs]: I'm assuming putting `[]` here is fine because its only being used after type checking phase.

Comment (judah) [reply src/GHC/SourceGen/Syntax/Internal.hs]: Yeah, this approach seems reasonable to me.

**Pull request merged**